share/ipld/nmt_adder.go

```diff
@@ -151,6 +159,18 @@ func (a *ProofsAdder) VisitFn() nmt.NodeVisitorFn {
 	return a.visitInnerNodes
 }
 
+// Purge removed proofs from ProofsAdder allowing GC to collect the memory
+func (a *ProofsAdder) Purge() {
+	if a == nil {
+		return
+	}
+
+	a.lock.RLock()
+	defer a.lock.RUnlock()
```

Comment:
this should not be a read lock

Reply:
There is also not so much value in read lock here. We write it once and use it everywhere. No simultaneous writes are possible. Also, visitFn is called sequentially by NMT.